plugins/websocket_server/WebsocketServer.cc

```diff
@@ -40,22 +40,152 @@ using namespace ignition::launch;
 /// \return A string that is the frame header.
 #define BUILD_MSG(_op, _topic, _type, _payload) (BUILD_HEADER(_op, _topic, _type) + _payload)
 
-int rootCallback(struct lws *_wsi,
-                 enum lws_callback_reasons _reason,
-                 void * /*user*/,
-                 void * _in,
-                 size_t /*len*/)
-{
+/// \brief Gets the websocket server from the lws protocol passed to handler.
+/// \attention The protocol should be defined with a reference to a websocket
+/// server in the `user` field.
+/// \param[in] _wsi lws connection.
+/// \return A pointer to the websocket server assigned to the protocol.
+WebsocketServer *get_server(struct lws *_wsi) {
   WebsocketServer *self = nullptr;
 
   // Get the protocol definition for this callback
-  lws_protocols *protocol = const_cast<lws_protocols*>(
-      lws_get_protocol(_wsi));
+  lws_protocols *protocol = const_cast<lws_protocols*>(lws_get_protocol(_wsi));
 
   // It's possible that the protocol is null.
   if (protocol)
     self = static_cast<WebsocketServer*>(protocol->user);
 
+  return self;
+}
+
+/// \brief Sets HTTP response status code and writes Content-Type and
+/// Content-Lenght HTTP headers.
+/// \param[in] _wsi lws connection.
+/// \param[in] status_code Status code.
+/// \param[in] content_type Content mime-type.
+/// \param[in] content_length Size of the body in bytes.
```

Comment:
Add documentation about return value.

Reply:
Addressed in https://github.com/ignitionrobotics/ign-launch/pull/49/commits/1fee42e8a1c08937f8610a1581d5926efdb396f1.